kernel/aster-nix/src/syscall/arch/x86.rs

```diff
@@ -0,0 +1,245 @@
+    clock_gettime::sys_clock_gettime,
+    clock_nanosleep::sys_clock_nanosleep,
+    clone::sys_clone,
+    close::sys_close,
+    connect::sys_connect,
+    dup::{sys_dup, sys_dup2},
+    epoll::{sys_epoll_create, sys_epoll_create1, sys_epoll_ctl, sys_epoll_wait},
+    execve::{sys_execve, sys_execveat},
+    exit::sys_exit,
+    exit_group::sys_exit_group,
+    fcntl::sys_fcntl,
+    fork::sys_fork,
+    fsync::sys_fsync,
+    futex::sys_futex,
+    getcwd::sys_getcwd,
+    getdents64::sys_getdents64,
+    getegid::sys_getegid,
+    geteuid::sys_geteuid,
+    getgid::sys_getgid,
+    getgroups::sys_getgroups,
+    getpeername::sys_getpeername,
+    getpgrp::sys_getpgrp,
+    getpid::sys_getpid,
+    getppid::sys_getppid,
+    getrandom::sys_getrandom,
+    getresgid::sys_getresgid,
+    getresuid::sys_getresuid,
+    getsid::sys_getsid,
+    getsockname::sys_getsockname,
+    getsockopt::sys_getsockopt,
+    gettid::sys_gettid,
+    gettimeofday::sys_gettimeofday,
+    getuid::sys_getuid,
+    ioctl::sys_ioctl,
+    kill::sys_kill,
+    link::{sys_link, sys_linkat},
+    listen::sys_listen,
+    lseek::sys_lseek,
+    madvise::sys_madvise,
+    mkdir::{sys_mkdir, sys_mkdirat},
+    mmap::sys_mmap,
+    mprotect::sys_mprotect,
+    munmap::sys_munmap,
+    open::{sys_open, sys_openat},
+    pause::sys_pause,
+    pipe::{sys_pipe, sys_pipe2},
+    poll::sys_poll,
+    prctl::sys_prctl,
+    pread64::sys_pread64,
+    prlimit64::sys_prlimit64,
+    read::sys_read,
+    readlink::{sys_readlink, sys_readlinkat},
+    recvfrom::sys_recvfrom,
+    rename::{sys_rename, sys_renameat},
+    rmdir::sys_rmdir,
+    rt_sigaction::sys_rt_sigaction,
+    rt_sigprocmask::sys_rt_sigprocmask,
+    rt_sigreturn::sys_rt_sigreturn,
+    sched_yield::sys_sched_yield,
+    select::sys_select,
+    sendto::sys_sendto,
+    set_get_priority::{sys_get_priority, sys_set_priority},
+    set_robust_list::sys_set_robust_list,
+    set_tid_address::sys_set_tid_address,
+    setfsgid::sys_setfsgid,
+    setfsuid::sys_setfsuid,
+    setgid::sys_setgid,
+    setgroups::sys_setgroups,
+    setpgid::sys_setpgid,
+    setregid::sys_setregid,
+    setresgid::sys_setresgid,
+    setresuid::sys_setresuid,
+    setreuid::sys_setreuid,
+    setsid::sys_setsid,
+    setsockopt::sys_setsockopt,
+    setuid::sys_setuid,
+    shutdown::sys_shutdown,
+    sigaltstack::sys_sigaltstack,
+    socket::sys_socket,
+    socketpair::sys_socketpair,
+    stat::{sys_fstat, sys_fstatat, sys_lstat, sys_stat},
+    statfs::{sys_fstatfs, sys_statfs},
+    symlink::{sys_symlink, sys_symlinkat},
+    sync::sys_sync,
+    syscall_dispatch, syscall_handler,
+    tgkill::sys_tgkill,
+    time::sys_time,
+    truncate::{sys_ftruncate, sys_truncate},
+    umask::sys_umask,
+    uname::sys_uname,
+    unlink::{sys_unlink, sys_unlinkat},
+    utimens::sys_utimensat,
+    wait4::sys_wait4,
+    waitid::sys_waitid,
+    write::sys_write,
+    writev::sys_writev,
+};
+
+syscall_dispatch! {
```

Comment:
To be precise, rename this macro to `impl_syscall_nums_and_syscall_dispatch_func`.